include/deal.II/base/subscriptor.h

```diff
@@ -92,6 +92,15 @@ class Subscriptor
    */
   Subscriptor &operator = (const Subscriptor &);
 
+#ifdef DEAL_II_WITH_CXX11
+  /**
+   * Move assignment operator.
+   *
+   * Asserts that the counter for the moved object is zero.
+   */
+  Subscriptor &operator = (Subscriptor &&);
```

Comment:
I tried to avoid adding this, but it was the best way I could come up with to change the move assignment operator for `Vector` ([here](https://github.com/dealii/dealii/pull/2503/files#diff-4741b74df28de226310a2345d233d906R1131)).

It will also be called under the hood by any default move assignment operators for objects that inherit from `Subscriptor`. Without a move assignment operator for subscriptor, the default will use the copy assignment instead. This might not give the desired error message in debug mode when something goes wrong.